src/Raven.Server/Web/Studio/Processors/BucketsHandlerProcessorForGetBucket.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Server.Documents;
+using Raven.Server.Documents.Sharding;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils;
+using Raven.Server.Web.Http;
+using NotImplementedException = System.NotImplementedException;
+
+namespace Raven.Server.Web.Studio.Processors
+{
+    internal class BucketsHandlerProcessorForGetBucket : AbstractBucketsHandlerProcessorForGetBucket<DatabaseRequestHandler, DocumentsOperationContext>
+    {
+        public BucketsHandlerProcessorForGetBucket([NotNull] DatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+        
+        protected override ValueTask<BucketInfo> GetBucketInfo(DocumentsOperationContext context, int bucket)
+        {
+            using (context.OpenReadTransaction())
+            {
+                var database = ShardedDocumentDatabase.CastToShardedDocumentDatabase(RequestHandler.Database);
+                var docs = database.ShardedDocumentsStorage.GetDocumentsByBucketFrom(context, bucket, etag: 0);
+                var bucketStats = ShardedDocumentsStorage.GetBucketStatisticsFor(context, bucket);
+
+                if (bucketStats == null)
```

Comment:
Under what scenario is this _possible_ ?

Reply:
When we try to get a bucket that doesn't exist

Reply:
In that case, how can it have any docs?

Reply:
It shouldn't, but we added it to check just in case there is a bug